src/transformStateWithClones.js

```diff
@@ -7,7 +7,39 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let currentState = { ...state };
+  const history = [];
+
+  for (const action of actions) {
+    switch (action.type) {
```

Comment:
```suggestion
  for (const action of actions) {
  const { type, extraData, keysToRemove } = action
    switch (type) {
    ...
```
Always use destruction